### PR TITLE
REQ-627 set up feature branch

### PR DESCRIPTION
**Pull request by lindig**
This series commits adds support for NVIDIA SR-IOV GPUs. It depends on corresponding changes in cen-api.git and xcp-idl.git.

### CODE REVIEW

File: xc/device.ml [hunk 1]
```diff
@@ -1267,21 +1272,22 @@ module PCI = struct
 
 
   let add ~xc ~xs ~hvm pcidevs domid =
+    let addr {host; guest=_} = host in
```
Comment (robhoes) [inline xc/device.ml]: Maybe call this function `host_addr` to avoid confusion?

File: xc/device.ml [hunk 2]
```diff
@@ -1141,6 +1141,11 @@ module PCI = struct
     driver: string;
   }
 
+  type device' =
+    { host:   Xenops_interface.Pci.address
+    ; guest:  int * Xenops_interface.Pci.address option
```
Comment (robhoes) [inline xc/device.ml]: How about defining a type name for the `int` to make it clearer what it represents (e.g. `type index = int`)?

File: xc/device.ml [hunk 3]
```diff
@@ -1144,6 +1144,7 @@ module PCI = struct
   type device' =
     { host:   Xenops_interface.Pci.address
     ; guest:  int * Xenops_interface.Pci.address option
+    ; qmp_add: bool (** false: don't issue Device_add commend *)
```
Comment (robhoes) [inline xc/device.ml]: "command"

**Comment by lindig** [comment]
I have addressed the comments in the `fixup!` commit which I would squash before merging.

**Pull request merged**